src/Controller/Admin/DataObject/DataObjectController.php

```diff
@@ -1163,6 +1164,9 @@ private function executeUpdateAction(DataObject $object, mixed $values): array
         }
 
         $key = $values['key'] ?? null;
+        if ($key) {
+            $key = Service::getValidKey($key, 'object');
+        }
```

Comment:
```suggestion
```
just found out that we handle it on client side https://github.com/pimcore/admin-ui-classic-bundle/blob/1976b8f869ad450e1b3d2b9e4aebcc47f3487e7e/public/js/pimcore/elementservice.js#L459

Reply:
is client-side enough? i guess we should also handle it on the server or?

Reply:
yes, right.